File: Anul 2/Metode Avansate de Programare/Seminarii/sem11-taskrunner/taskrunner/taskrunner/factory/TaskContainerFactory.cs
using taskrunner.container;

namespace taskrunner.factory;

public class TaskContainerFactory : IFactory
{
    private static TaskContainerFactory instance = null;

    private TaskContainerFactory()
    {
    }

    public static TaskContainerFactory GetInstance()
    {
        if (instance == null)
            instance = new TaskContainerFactory();

        return instance;
    }

    public IContainer CreateContainer(Strategy strategy)
    {
        if (strategy == Strategy.LIFO)
            return new StackContainer();
        else
            return new QueueContainer();
    }
}

File: Anul 2/Metode Avansate de Programare/Laboratoare/lab-1-Mothyque/src/factory/TaskContainerFactory.java
package factory;

import domain.Container;
import domain.QueueContainer;
import domain.StackContainer;
import domain.Strategy;

public class TaskContainerFactory implements Factory
{

    //private static TaskContainerFactory instance = new TaskContainerFactory();

    private TaskContainerFactory() {}

    private static TaskContainerFactory instance = null;
    public static TaskContainerFactory getInstance()
    {
        if (instance != null)
        {
            return instance;
        }
        else
        {
            instance = new TaskContainerFactory();
            return instance;
        }
    }

    @Override
    public Container createContainer(Strategy strategy)
    {
        if (strategy == Strategy.FIFO)
        {
            return new QueueContainer();
        }
        else if (strategy == Strategy.LIFO)
        {
            return new StackContainer();
        }
        return null;
    }
}